scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -2111,14 +2149,16 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
     def appendMod(mods: List[Mod], mod: Mod): List[Mod] = {
       def validate() = {
         if (isLocal && !mod.tokens.head.is[LocalModifier]) syntaxError("illegal modifier for a local definition", at = mod)
-        mods.foreach(m => if (m.productPrefix == mod.productPrefix) syntaxError("repeated modifier", at = mod))
+        if (!mod.isInstanceOf[Mod.Quasi]) mods.foreach(m => if (m.productPrefix == mod.productPrefix) syntaxError("repeated modifier", at = mod))
         if (mod.hasAccessBoundary) mods.filter(_.hasAccessBoundary).foreach(mod => syntaxError("duplicate access qualifier", at = mod))
       }
       validate()
       mods :+ mod
     }
+    def continueLoop = ahead(token.is[`:`] || token.is[`=`] || token.is[EOF])
     def loop(mods: List[Mod]): List[Mod] = token match {
-      case _: Modifier         => loop(appendMod(mods, modifier()))
+      case _: Unquote => if (continueLoop) mods else loop(appendMod(mods, modifier()))
+      case _: Modifier | _: Ellipsis => loop(appendMod(mods, modifier()))
```

Comment:
I think it is possible, but uniform parsing of Unquotes and Ellipses itself is another refactoring issue. And it requires careful usage analysis (there are much more such cases).